spiffworkflow-backend/src/spiffworkflow_backend/models/message_instance.py

```diff
@@ -45,21 +46,22 @@ class MessageInstanceModel(SpiffworkflowBaseDBModel):
     __tablename__ = "message_instance"
 
     id: int = db.Column(db.Integer, primary_key=True)
-    process_instance_id: int = db.Column(ForeignKey(ProcessInstanceModel.id), nullable=False)  # type: ignore
-    message_model_id: int = db.Column(ForeignKey(MessageModel.id), nullable=False)
-    message_model = relationship("MessageModel")
-    message_correlations_message_instances = relationship(
-        "MessageCorrelationMessageInstanceModel", cascade="delete"
-    )
-
+    process_instance_id: int = db.Column(ForeignKey(ProcessInstanceModel.id), nullable=True)  # type: ignore
+    name: str = db.Column(db.String(255))
     message_type: str = db.Column(db.String(20), nullable=False)
-    payload: str = db.Column(db.JSON)
+    payload: dict = db.Column(db.JSON)
     status: str = db.Column(db.String(20), nullable=False, default="ready")
+    user_id: int = db.Column(ForeignKey(UserModel.id), nullable=False)  # type: ignore
```

Comment:
messages are normally processed in the background so I'm not sure who the user will be unless it's the process instance initiator.

Reply:
It is defaulting to exactly that - or in the case of an API call the user_id is set to who made that API Call -- in this way we can determine how a process instance was started.